src/main/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncher.java

```diff
@@ -0,0 +1,245 @@
+		this.properties = properties;
+		this.chronos = chronos;
+	}
+
+	@Override
+	public String launch(AppDeploymentRequest request) {
+		String jobName = createDeploymentId(request);
+		String image = null;
+		try {
+			image = request.getResource().getURI().getSchemeSpecificPart();
+		} catch (IOException e) {
+			throw new IllegalArgumentException("Unable to get URI for " + request.getResource(), e);
+		}
+		logger.info("Using Docker image: " + image);
+		DockerJob job = new DockerJob();
+		job.setName(jobName);
+		List<Map<String, String>> envVars = new ArrayList<>();
+		Map<String, String> springApplicationJson = createSpringApplicationJson(request);
+		if (springApplicationJson.size() > 0) {
+			envVars.add(springApplicationJson);
+		}
+		logger.info("Using env: " + envVars);
+		if (envVars.size() > 0) {
+			job.setEnvironmentVariables(envVars);
+		}
+		job.setShell(false);
+		job.setCommand("");
+		List<String> args = createCommandArgs(request);
+		if (args.size() > 0) {
+			job.setArguments(args);
+		}
+		job.setSchedule("R1//P");
+		job.setRetries(1);
+		DockerContainer container = new DockerContainer();
+		container.setImage(image);
+		job.setContainer(container);
+		Double cpus = deduceCpus(request);
+		Double memory = deduceMemory(request);
+		job.setCpus(cpus);
+		job.setMem(memory);
+		if (StringUtils.hasText(properties.getOwnerEmail())) {
+			job.setOwner(properties.getOwnerEmail());
+		}
+		if (StringUtils.hasText(properties.getOwnerName())) {
+			job.setOwnerName(properties.getOwnerName());
+		}
+		if (properties.getUris() != null && properties.getUris().length > 0) {
+			job.setUris(Arrays.asList(properties.getUris()));
+		}
+		try {
+			if (logger.isDebugEnabled()) {
+				logger.debug("Launching Job with definition:\n" + job.toString());
+			}
+			chronos.createDockerJob(job);
+		} catch (ChronosException e) {
+			logger.error(e.getMessage(), e);
+			throw new IllegalStateException(String.format("Error while creating job '%s'", jobName), e);
+		}
+		return jobName;
+	}
+
+	@Override
+	public void cancel(String id) {
+		try {
+			chronos.deleteJobTasks(id);
+		} catch (ChronosException e) {
+			throw new IllegalStateException(String.format("Error while canceling job '%s'", id), e);
+		}
+	}
+
+	@Override
+	public TaskStatus status(String id) {
+		String csv = null;
+		try {
+			csv = chronos.getGraphCsv();
+		} catch (ChronosException e) {
+			throw new IllegalStateException(String.format("Error while retrieving graph"), e);
+		}
+		List<Job> list = null;
+		try {
+			list = chronos.getJobs();
+		} catch (ChronosException e) {
+			throw new IllegalStateException(String.format("Error while retrieving jobs"), e);
+		}
+		Job job = null;
+		for (Job j : list) {
+			if (j.getName().equals(id)) {
+				job = j;
+			}
+		}
+		TaskStatus status = buildTaskStatus(properties, id, job, csv);
+		logger.debug(String.format("Status for task: %s is %s", id, status));
+
+		return status;
+	}
+
+	//	@Override
+	public void cleanup(String... id) {
```

Comment:
We need to somehow use this method when destroying the tasks? Currently, the behaviour of task definition controller is to just delete the task definition repository entry. I think it should as well call `cleanup`. May be we need to have some contract in `TaskLauncher` for `cleanup`? \cc @trisberg @mminella @cppwfs